### ISSUE TITLE
Destructuring possible null value with --downlevelIteration and any ES --lib ignores errors

### ISSUE THREAD

**Issue by Theomg** [microsoft/TypeScript#35497]
**TypeScript Version:**  3.7.3 and 3.8.0-dev.20191204

**Search Terms:**
TS2488
TS2548
downlevelIteration lib destructuring error

**Code**

```ts
(data: number[] | null) => {
  const [value] = data;
};
```

**Expected behavior:**
Using array destructuring on `data` should cause TypeScript error.

**Actual behavior:**
When using compilerOptions downlevelIteration and lib with any es option (es2016, esnext, etc) typescript presents no errors when array destructuring a possibly null variable.

**Playground Link:**
Due to using `lib` it has to be on code sandbox
No error due to using downlevelIteration and lib: https://codesandbox.io/s/zen-galois-dguz9
Error as expected: https://codesandbox.io/s/summer-dust-fs4r7

**Related Issues:**
Couldn't find related issues.

**Comment by andrewbranch** [comment]
I’ve been playing with this one for a while now—fixing it is pretty easy, but  the fix causes a seemingly unrelated change that isn’t _wrong_ per se but is probably undesirable and definitely breaking. I think I can add a special case to keep that from happening, but it makes me nervous that other things will break unexpectedly. I’ll keep messing with it today but this is probably going to get bumped to 4.1.

**Comment by andrewbranch** [comment]
@rbuckton, I’d like to get your take on this and https://github.com/microsoft/TypeScript/pull/40350. I could bring it to the design meeting this week, but maybe you have some thoughts first.
